960-api-services/src/main/java/io/harness/aws/AwsClientImpl.java

```
@@ -505,6 +514,37 @@ public Map<String, String> listIAMRoles(AwsInternalConfig awsInternalConfig) {
     return emptyMap();
   }
 
+  @Override
+  public String getEcrImageUrl(AwsInternalConfig awsConfig, String registryId, String region, String imageName) {
+    Repository repository = getRepository(awsConfig, registryId, region, imageName);
+    return repository != null ? repository.getRepositoryUri() : null;
```

Should we instead throw an error saying that unable to find repository for region, registry and image name?

---

Otherwise let's use Optional instead of returning null (still this case should be handled at some point of time, it's not valid if the repository url is null)

---

Done